### DISCUSSION TITLE
Variables: Preview? Interim alternative? (Or just be patient?)

### DISCUSSION THREAD

**Discussion by ghost** [retypeapp/retype#78] [Q&A] [answered]
1. I'm interested in how variables will look in v1.9.0. Do you have any preview or samples to illustrate? Coming from MkDocs-like SWGs so assuming something simple like:

     ```md
     ---
     myvar: 'value'
     ---

     # H1
     My var is {{myvar}}
    ```

    with perhaps a way of loading larger global variable files.

2. Are there any variable-like workarounds I can use now?

**Comment by geoffreymcgill** [comment, marked as answer]
Yes, exactly the same configuration and usage as you provided is the plan. The syntax for the placement of the variable on the page would be `{{ myvar }}`.

We will also include functionality to configure variables in the `retype.json` file, then reference those `variables` on any Page. Current proposal for configuration within `retype.json` would be something like the following:

```json
{
    "variables": {
        "my-var-name": "custom-value",
        "another-var": 55
    }
}
```

Plus also access to values from the `project`, `page`, `environment`, and likely others.

> Are there any variable-like workarounds I can use now?

Unfortunately, no. Nothing that I can think of right now. We will try our best to release as soon as possible, although `v1.9.0` is currently scheduled for mid-Q3 and that timeframe still seems likely.

**Reply by ghost** [reply]
Good answer, thanks. So I'll just "be patient".
